### PR TITLE
new melee weapons

### PR DESCRIPTION
**Pull request by Tarlan2**
SS14-test
Added two new melee weapons: the combat knife and the fire axe. 
Both got tested, both work.
The only problem is that the inhand sprite makes the axe look like the size of the hatchet.

:cl:
- add: Adds combat knife and fire axe.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+- type: entity
+  name: fireaxe
+  parent: BaseItem
+  id: Firexe
+  description: A large, robust axe. Can pry open doors and skulls alike.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/fireaxe.rsi
+    state: icon
+
```
Comment (SweptWasTaken) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml:10]: Unindent

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml [hunk 2]
```diff
@@ -48,3 +48,21 @@
...
+  name: combat knife
+  parent: BaseKnife
+  id: CombatKnife
+  description: A deadly blade intended for melee confrontations.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/combat_knife.rsi
+    size: 2
+    state: icon
+
```
Comment (SweptWasTaken) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml:62]: Unindent

File: Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                [
+                    1.0
+                ],
+                [
+                    1.0
+                ],
+                [
+                    1.0
+                ]
+            ]
```
Comment (SweptWasTaken) [inline Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json:26]: Delete entire delays section. (Remember to remove the command from directions.)

File: Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        },
+        {
+            "name": "inhand-right",
+            "directions": 4,
+            "delays": [
```
Comment (SweptWasTaken) [inline Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json:31]: Same as above

**Review by SweptWasTaken — CHANGES_REQUESTED** [review]
Knife sprite looks like garbage.

File: Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json (hunk 4 above)
Comment (Tarlan2) [reply Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json]: should I do the same for the knife?

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+- type: entity
+  name: fireaxe
+  parent: BaseItem
+  id: Fireaxe
+  description: A large, robust axe. Can pry open doors and skulls alike.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/fireaxe.rsi
+    state: icon
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml:9]: netsync false unless there's a good reason not to.

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+  - type: Fireaxe
+  - type: ItemCooldown
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml:12]: Why does it need this?

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  - type: MeleeWeapon
+    damage: 30
+  - type: Item
+    size: 24
+    sprite: Objects/Weapons/Melee/fireaxe.rsi
+    prefix: inhand
+  - type: LoopingSound
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml:19]: Doesn't need loopingsound

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml [hunk 8]
```diff
@@ -48,3 +48,21 @@
...
     prefix: inhand
+
+- type: entity
+  name: combat knife
+  parent: BaseKnife
+  id: CombatKnife
+  description: A deadly blade intended for melee confrontations.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/combat_knife.rsi
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/knife.yml:59]: As above

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  name: fireaxe
+  parent: BaseItem
+  id: Fireaxe
+  description: A large, robust axe. Can pry open doors and skulls alike.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/fireaxe.rsi
+    state: icon
+
+  - type: Fireaxe
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml:11]: This component doesn't exist?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by SweptWasTaken** [comment]
Like, I get that you're copy-pasting from somewhere but you are building this right? You must have to have gotten the fire-axe ingame. Did you not see the error in the console saying `Unknown component 'Fireaxe' in prototype Fireaxe!`.

Copy-pasting to fill out content is fine but you can't just copy/paste stuff BLINDLY like this.

Also you didn't even do what I told you to do. I told you to remove the spacing on the components and you only removed one line. I showed you this in discord.

Please give [this guide](https://hackmd.io/@ss14/yaml) a read before making another PR. (I'll know if you haven't).

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml (hunk 5 above)
Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/fireaxe.yml:9]: do they need netsyncing or what

File: Resources/Textures/Objects/Weapons/Melee/combat_knife.rsi/meta.json [hunk 10]
```diff
@@ -0,0 +1,22 @@
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "Taken/modified from tgstation at commit https://github.com/tgstation/tgstation/pull/30680/commits/a494d500b42bc8c7275c57219e08e65b825e94f6",
```
Comment (PaulRitter) [inline Resources/Textures/Objects/Weapons/Melee/combat_knife.rsi/meta.json:4]: is this correct? i thought you sprited this yourself?

**Review by PaulRitter — CHANGES_REQUESTED** [review]
knife could be a bit smaller imo

File: Resources/Textures/Objects/Weapons/Melee/combat_knife.rsi/meta.json (hunk 10 above)
Comment (Tarlan2) [reply Resources/Textures/Objects/Weapons/Melee/combat_knife.rsi/meta.json:4]: Yeah, I used the tg combat knife as an inspiration.  Also, already made the sprite smaller by reducing the handle and having it centered.

**Comment by ike709** [comment]
That knife is comically large.

File: Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json [hunk 11]
```diff
@@ -0,0 +1,22 @@
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
```
Comment (Zumorica) [inline Resources/Textures/Objects/Weapons/Melee/fireaxe.rsi/meta.json:3]: Goon assets are all CC-BY-NC-SA-3.0.
```suggestion
    "license": "CC-BY-NC-SA-3.0",
```

**Pull request merged**